packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+
+module.exports = compile
+
+function compile (node) {
+  if (node === null || typeof node === 'number' || typeof node === 'boolean' || typeof node === 'string') {
+    return JSON.stringify(node)
+  }
+
+  const [type, value] = Object.entries(node)[0]
+
+  if (type === 'not') {
+    return `!(${compile(value)})`
+  } else if (type === 'isEmpty') {
+    return `(${compile(value)}).length === 0`
+  } else if (type === 'isDefined') {
+    return `(() => {
+        try {
+          ${value}
```

Comment:
should we add `;` or `()` around the value here?

Reply:
Isn't this always a reference to a variable? If so, it shouldn't be required, or?

Reply:
I see your point. 99% of the time is will just be a reference. it could be a complex one like map[index].field but those are get translated into a method call with () around it.